src/main/java/io/cdap/plugin/jira/source/common/JiraClient.java

```diff
@@ -39,12 +46,36 @@
   private static final ImmutableSet<String> MINIMAL_FIELDS_SET = ImmutableSet.of("project", "summary", "issuetype",
                                                                                  "created", "updated", "status");
 
+  private static final Pattern ORDER_BY_PATTERN = Pattern.compile(
+    "ORDER(\\s)+BY(\\s)+(\\w+)(,\\s*\\w+)*(\\s+ASC|\\s+DESC)*", Pattern.CASE_INSENSITIVE);
+
+  private static final Logger LOG = LoggerFactory.getLogger(JiraClient.class);
+
   private final JiraRestClient restClient;
-  private final BaseJiraSourceConfig config;
+  private final JiraSourceConfig config;
+  private final String startFromCreateDate;
+  private final String queryOrderByPostfix;
+  private final String trackingField;
 
-  public JiraClient(BaseJiraSourceConfig config) {
+  public JiraClient(JiraSourceConfig config) {
+    this(config, null, false);
+  }
+
+  public JiraClient(JiraSourceConfig config, @Nullable ZonedDateTime startFromCreateDate, Boolean trackUpdates) {
     this.config = config;
 
+    if (startFromCreateDate != null) {
```

Comment:
wouldn't it look better with 2 constructors instead of if?

Reply:
Discussed offline. The second constructor can have runtime null argument that's why it's needed.